### PR TITLE
Catch non-integer bucket/count values.

### PR DESCRIPTION
**Pull request by brian-brazil**
@roidelapluie

### CODE REVIEW

File: prometheus_client/openmetrics/parser.py [hunk 1]
```diff
@@ -562,10 +562,16 @@ def build_metric(name, documentation, typ, unit, samples):
 
             if typ == 'stateset' and name not in sample.labels:
                 raise ValueError("Stateset missing label: " + line)
-            if (typ in ['histogram', 'gaugehistogram'] and name + '_bucket' == sample.name
+            if (name + '_bucket' == sample.name
```
Comment (csmarchbanks) [inline prometheus_client/openmetrics/parser.py:565]: I am curious here around removing the type for this check? Looking at the OM spec it is only a SHOULD that `_bucket`/`_count` series are not used by other types so I would have expected the type checks to remain for all of these.

Comment (brian-brazil) [reply prometheus_client/openmetrics/parser.py:565]: It is redundant, as it is a MUST that they're not used by others as a suffix so any deviations will be caught elsewhere. That bit of the spec is around naming metrics, not use of suffixes.

Comment (csmarchbanks) [reply prometheus_client/openmetrics/parser.py:565]: Ahh right, they would be potential clashes, that makes sense. Thanks.

**Review by csmarchbanks — APPROVED** [review]

**Review by roidelapluie — APPROVED** [review]
Thanks!

**Pull request merged**